plugin/src/main/java/dev/regadas/trino/pubsub/listener/encoder/AvroQueryEventEncoder.java

```diff
@@ -0,0 +1,38 @@
+package dev.regadas.trino.pubsub.listener.encoder;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.avro.AvroMapper;
+import com.fasterxml.jackson.dataformat.avro.AvroSchema;
+import com.fasterxml.jackson.dataformat.avro.jsr310.AvroJavaTimeModule;
+import com.fasterxml.jackson.dataformat.avro.schema.AvroSchemaGenerator;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.google.common.annotations.VisibleForTesting;
+import dev.regadas.trino.pubsub.listener.event.QueryEvent;
+
+public class AvroQueryEventEncoder implements QueryEventEncoder {
```

Comment:
```suggestion
public class AvroQueryEventEncoder implements Encoder<QueryEvent> {
```
I think this makes more sense... `QueryEventEncoder` acts more as a helper